src/spec/context.cr

```diff
@@ -330,6 +368,11 @@ module Spec
       Spec.formatters.each(&.pop)
     end
 
+    def run
+      run
+      yield
+    end
```

Comment:
What is this for?

Reply:
I'm not entirely sure anymore. I wrote this over a year ago. 🙂

I remember I needed something to send a notification of completion to the main fiber. I don't remember why I implemented it this way specifically, though.

Reply:
Is this actually used? I don't see where.

Reply:
I've had to update this PR several times since it's been opened due to implementation changes in `Spec`, so I'm not 100% sure which changes were made when and why, but I definitely did use it at one point.

Looks like I used it in the beginning when [there was a distinction between sync vs async specs](https://github.com/crystal-lang/crystal/pull/8944/commits/522d9ac5f8b611bb4060db4c1a47b34af62815c0#diff-11054ce691124ab4ac3b7571b3f204de21d242ce54d649424ac4a30da6bbffacR55-R57) (before I added the `-j` option).